sbtplugin/src/main/scala/scala/scalanative/sbtplugin/ScalaNativePluginInternal.scala

```diff
@@ -109,7 +109,8 @@ object ScalaNativePluginInternal {
     nativeSharedLibrary := false,
 
     nativeLink := {
-      val entry         = (selectMainClass in Compile).value.get.toString + "$"
+      val mainClass     = (selectMainClass in Compile).value.getOrElse(sys.error("No main class detected."))
```

Comment:
Please use `throw new MessageOnlyException` as seen above instead of `sys.error`.